### PR TITLE
chore: config module.

### PR DESCRIPTION
**Pull request by tokers**
Re designed a config module, supporting initializing from command line optoins and files.
Now this module is not in use. Will be used in the near future.

### CODE REVIEW

File: cmd/ingress/ingress.go [hunk 1]
```diff
@@ -71,19 +98,22 @@ func NewIngressCommand() *cobra.Command {
 			}()
 
 			router := pkg.Route()
-			err := http.ListenAndServe(":8080", router)
+			err = http.ListenAndServe(":8080", router)
```
Comment (gxthrj) [inline cmd/ingress/ingress.go:101]: The `:8080` need to use `config.HTTPListen` instead.

File: pkg/config/config.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+	"io/ioutil"
+	"strings"
+	"time"
+
+	"github.com/api7/ingress-controller/pkg/types"
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	_minimalResyncInterval = time.Second
```
Comment (gxthrj) [inline pkg/config/config.go:29]: The Frequency  mini interval of resync 1 per second is too high, how about 1min, the same as default interval.

Comment (tokers) [reply pkg/config/config.go:29]: I think 1min is too large, what about 30s.

File: cmd/ingress/ingress.go (hunk 1 above)
Comment (tokers) [reply cmd/ingress/ingress.go:101]: Sure, will be changed in next PR :).

This PR will not introduce any codes that use the new config.

Comment (gxthrj) [reply cmd/ingress/ingress.go:101]: OK, understanding.

File: pkg/config/config.go (hunk 2 above)
Comment (gxthrj) [reply pkg/config/config.go:29]: ok

Comment (tokers) [reply pkg/config/config.go]: Changed.

**Review by membphis — COMMENTED** [review]
LGTM

**Review by gxthrj — APPROVED** [review]
LGTM

**Pull request merged**